### PR TITLE
CIV-13490 and civ-13491 intermediate DQs

### PR DESCRIPTION
**Pull request by drummondjm**
https://tools.hmcts.net/jira/browse/CIV-13490
https://tools.hmcts.net/jira/browse/CIV-13491

Intermediate claim respond to claim as defendant and then as claimant.

Addition of new screen for fix recoverable costs.

image

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

**Review by LeonardoPalmeiro — DISMISSED** [review]

**Review by sankaviv1 — APPROVED** [review]

File: ccd-definition/ComplexTypes/DQ/FixedRecoverableCostsIntermediate.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    "ID": "FixedRecoverableCostsIntermediate",
+    "ListElementCode": "isSubjectToFixedRecoverableCostRegime",
+    "FieldType": "YesOrNo",
+    "ElementLabel": "Is this claim subject to the Fixed Recoverable Cost Regime requiring the court to determine a complexity band?",
+    "SecurityClassification": "Public",
+    "Searchable": "N"
+  },
+  {
+    "ID": "FixedRecoverableCostsIntermediate",
+    "ListElementCode": "agreedWithOtherParty",
```
Comment (sherlynkhaw) [inline ccd-definition/ComplexTypes/DQ/FixedRecoverableCostsIntermediate.json:20]: don't think a new name is needed for this, we already have a field that is similar could we change it to `complexityBandingAgreed`

File: e2e/fixtures/events/claimantResponseIntermediateClaimSpec.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+              explainedToClient: ['CONFIRM'],
+              oneMonthStayRequested: 'No',
+              reactionProtocolCompliedWith: 'No',
+              reactionProtocolNotCompliedWithReason: 'test'
+            }
+          },
+          FixedRecoverableCostsIntermediate: {
+            applicant1DQFixedRecoverableCostsIntermediate: {
+              isSubjectToFixedRecoverableCostRegime: 'Yes',
+              agreedWithOtherParty:'Yes',
```
Comment (sherlynkhaw) [inline e2e/fixtures/events/claimantResponseIntermediateClaimSpec.js:25]: complexityBandingAgreed

File: ccd-definition/CaseEventToComplexTypes/ClaimantResponseLRspec.json [hunk 3]
```diff
@@ -615,5 +615,60 @@
...
+    "ListElementCode": "isSubjectToFixedRecoverableCostRegime",
+    "ElementLabel": " ",
+    "FieldDisplayOrder": 2,
+    "DisplayContext": "MANDATORY"
+  },
+  {
+    "ID": "FixedRecoverableCostsIntermediate",
+    "CaseEventID": "CLAIMANT_RESPONSE_SPEC",
+    "CaseFieldID": "applicant1DQFixedRecoverableCostsIntermediate",
+    "ListElementCode": "agreedWithOtherParty",
```
Comment (sherlynkhaw) [inline ccd-definition/CaseEventToComplexTypes/ClaimantResponseLRspec.json:641]: complexityBandingAgreed

File: ccd-definition/CaseEventToComplexTypes/DefendantResponseLRspec.json [hunk 4]
```diff
@@ -1336,5 +1336,115 @@
...
+    "ListElementCode": "isSubjectToFixedRecoverableCostRegime",
+    "ElementLabel": " ",
+    "FieldDisplayOrder": 2,
+    "DisplayContext": "MANDATORY"
+  },
+  {
+    "ID": "FixedRecoverableCostsIntermediate",
+    "CaseEventID": "DEFENDANT_RESPONSE_SPEC",
+    "CaseFieldID": "respondent2DQFixedRecoverableCostsIntermediate",
+    "ListElementCode": "agreedWithOtherParty",
```
Comment (sherlynkhaw) [inline ccd-definition/CaseEventToComplexTypes/DefendantResponseLRspec.json:1417]: complexityBandingAgreed

File: ccd-definition/CaseEventToComplexTypes/DefendantResponseLRspec.json [hunk 5]
```diff
@@ -1336,5 +1336,115 @@
...
+    "ListElementCode": "isSubjectToFixedRecoverableCostRegime",
+    "ElementLabel": " ",
+    "FieldDisplayOrder": 2,
+    "DisplayContext": "MANDATORY"
+  },
+  {
+    "ID": "FixedRecoverableCostsIntermediate",
+    "CaseEventID": "DEFENDANT_RESPONSE_SPEC",
+    "CaseFieldID": "respondent1DQFixedRecoverableCostsIntermediate",
+    "ListElementCode": "agreedWithOtherParty",
```
Comment (sherlynkhaw) [inline ccd-definition/CaseEventToComplexTypes/DefendantResponseLRspec.json:1362]: complexityBandingAgreed

File: e2e/fixtures/events/claimantResponseIntermediateClaimSpec.js [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+      case 'PART_ADMISSION':
+        responseData.userInput = {
+          ...responseData.userInput,
+          RespondentResponse: {
+            applicant1ProceedWithClaim: 'Yes',
+          },
+          FixedRecoverableCostsIntermediate: {
+            applicant1DQFixedRecoverableCostsIntermediate: {
+              isSubjectToFixedRecoverableCostRegime: 'Yes',
+              agreedWithOtherParty:'Yes',
```
Comment (sherlynkhaw) [inline e2e/fixtures/events/claimantResponseIntermediateClaimSpec.js:131]: complexityBandingAgreed

File: e2e/tests/api_tests/multiIntermediateTrack/api_intermediate_spec_test.js [hunk 7]
```diff
@@ -4,17 +4,54 @@ const config = require('../../../config.js');
...
 
-Feature('CCD 1v1 API test spec intermediate  track @api-spec-multi-intermediate @api-nonprod');
+Feature('CCD 1v1 API test spec intermediate  track @api-spec-multi-intermediate');
 
 async function prepareClaim(api_spec, mpScenario) {
   await api_spec.createClaimWithRepresentedRespondent(config.applicantSolicitorUser, mpScenario, false, true, claimAmountPenniesIntermediate);
 }
 
-Scenario('1v1 full defence Intermediate claim Specified @api-nonprod-specified', async ({I, api_spec}) => {
+Scenario('1v1 FULL_DEFENSE Intermediate claim Specified @api-nonprod-specified', async ({api_spec}) => {
```
Comment (sherlynkhaw) [inline e2e/tests/api_tests/multiIntermediateTrack/api_intermediate_spec_test.js:14]: small spelling error FULL_DEFENCE

File: e2e/tests/api_tests/multiIntermediateTrack/api_multi_spec_test.js [hunk 8]
```diff
@@ -14,24 +13,44 @@ async function prepareClaim(api_spec, mpScenario) {
...
 });
 
 Scenario('1v2 full defence Multi claim Specified Different Solicitor', async ({api_spec}) => {
   const mpScenario = 'ONE_V_TWO';
   await prepareClaim(api_spec, mpScenario);
-  await api_spec.defendantResponse(config.defendantSolicitorUser, 'FULL_DEFENCE1', 'ONE_V_ONE_DIF_SOL',
-    'AWAITING_RESPONDENT_ACKNOWLEDGEMENT', false, true, claimAmountMulti);
-  await api_spec.defendantResponse(config.secondDefendantSolicitorUser, 'FULL_DEFENCE2', 'ONE_V_ONE_DIF_SOL',
-    'AWAITING_APPLICANT_INTENTION', false, true, claimAmountMulti);
+  await api_spec.defendantResponse(config.defendantSolicitorUser, 'FULL_DEFENCE1', 'ONE_V_ONE_DIF_SOL', 'AWAITING_RESPONDENT_ACKNOWLEDGEMENT', false, true, claimAmountMulti);
```
Comment (sherlynkhaw) [inline e2e/tests/api_tests/multiIntermediateTrack/api_multi_spec_test.js:44]: FULL_DEFENCE1 not sure if typo?

**Review by sherlynkhaw — CHANGES_REQUESTED** [review]

File: ccd-definition/ComplexTypes/DQ/FixedRecoverableCostsIntermediate.json (hunk 1 above)
Comment (drummondjm) [reply ccd-definition/ComplexTypes/DQ/FixedRecoverableCostsIntermediate.json:20]: Yeah remnant from when we were gonna use a bold label,  agreedWithOtherPartywould have been the actual yes/no with empty label and an additional agreedWithOtherPartyLabel for the actual bold label. No longer need bold text, so can just reuse that  complexityBandingAgreed now

File: e2e/tests/api_tests/multiIntermediateTrack/api_multi_spec_test.js (hunk 8 above)
Comment (drummondjm) [reply e2e/tests/api_tests/multiIntermediateTrack/api_multi_spec_test.js:44]: It's so it use the scenario in steps_LRspec.js, functionally the same as FULL_DEFENCE, but aligns with other 1v2 diff sol tests

   ONE_V_ONE_DIF_SOL: {
      FULL_DEFENCE1: data.DEFENDANT_RESPONSE('FULL_DEFENCE'),
      FULL_DEFENCE1_PBAv3: data.DEFENDANT_RESPONSE('FULL_DEFENCE', 'CREATE_CLAIM_SPEC_AFTER_PAYMENT'),
      FULL_DEFENCE1_PBAv3_Mediation: data.DEFENDANT_RESPONSE_1v2_Mediation('FULL_DEFENCE', 'CREATE_CLAIM_SPEC_AFTER_PAYMENT'),
      FULL_DEFENCE1_PBAv3_MULTI_CLAIM: data.DEFENDANT_RESPONSE_MULTI_CLAIM('FULL_DEFENCE', 'CREATE_CLAIM_SPEC_AFTER_PAYMENT'),

File: e2e/tests/api_tests/multiIntermediateTrack/api_intermediate_spec_test.js (hunk 7 above)
Comment (drummondjm) [reply e2e/tests/api_tests/multiIntermediateTrack/api_intermediate_spec_test.js:14]: Done

File: e2e/fixtures/events/claimantResponseIntermediateClaimSpec.js (hunk 6 above)
Comment (drummondjm) [reply e2e/fixtures/events/claimantResponseIntermediateClaimSpec.js:131]: same

File: ccd-definition/CaseEventToComplexTypes/DefendantResponseLRspec.json (hunk 4 above)
Comment (drummondjm) [reply ccd-definition/CaseEventToComplexTypes/DefendantResponseLRspec.json:1417]: same

File: ccd-definition/CaseEventToComplexTypes/ClaimantResponseLRspec.json (hunk 3 above)
Comment (drummondjm) [reply ccd-definition/CaseEventToComplexTypes/ClaimantResponseLRspec.json:641]: same

File: e2e/fixtures/events/claimantResponseIntermediateClaimSpec.js (hunk 2 above)
Comment (drummondjm) [reply e2e/fixtures/events/claimantResponseIntermediateClaimSpec.js:25]: same

**Review by sherlynkhaw — APPROVED** [review]

**Review by sherlynkhaw — APPROVED** [review]

**Pull request merged**